src/Poisson3D.cpp

```diff
@@ -0,0 +1,460 @@
+#include "Poisson3D.hpp"
+
+
+void
+Poisson3D::write_csv(const long int elapsed_time, int iterations) const{
+    std::ofstream csv_file;
+    csv_file.open("../results/results.csv", std::ios_base::app);
+
+    if (!csv_file.is_open()) {
+        std::cerr<<"Error! Cannot open csv file! ";
+        return ;
+    }
+
+    csv_file << extractFileName(mesh_file_name) << ","
+             << p_value <<","
+             << r <<","
+             << preconditioner_name << ","
+             << elapsed_time<<","
+             << iterations <<","
+             << symmetric <<","
+             << 0
+             << std::endl;
+    csv_file.close();
+
+    std::cout<< "CSV write successful."<<std::endl;
+}
+
+std::string
+Poisson3D::extractFileName(const std::string& filePath) {
+
+    size_t lastSlash = filePath.find_last_of("/\\");
```

Comment:
```suggestion
    const size_t lastSlash = filePath.find_last_of("/\\");
```
Always declare as const the variables whose value is not meant to change.